src/Core/Form/IdentifiableObject/DataHandler/FeatureFormDataHandler.php

```diff
@@ -57,7 +67,7 @@ public function create(array $data)
         /** @var FeatureId $featureId */
         $featureId = $this->commandBus->handle(new AddFeatureCommand(
             $data['name'],
-            $data['shop_association'] ?? []
+            $data['shop_association'] ?? $this->shopContext->getContextShopIds()
```

Comment:
We have more and more use cases like this one, I think there's room for improvement this could be handled by `ShopChoiceTreeType` that would automatically handle the multishop state, and would turn into a hidden field when it's disabled and automatically populate based on the shop context.

This way we could simplify all the related form data handler

Reply:
provided `'default_empty_data' => $this->shopContext->getContextShopIds()` in ShopTreeChoiceType, this seems to do the trick :+1: